packages/runtime-core/__tests__/components/Teleport.spec.ts

```diff
@@ -472,4 +472,30 @@ describe('renderer: teleport', () => {
     expect(dir.mounted).toHaveBeenCalledTimes(1)
     expect(dir.unmounted).toHaveBeenCalledTimes(1)
   })
+
+  //#3623
+  test('nested teleport should be removed when unmounted', () => {
+    const target = nodeOps.createElement('div')
+    const root = nodeOps.createElement('div')
+
+    render(
+      h(() => [
+        h(
+          Teleport,
+          { to: target },
+          h('div', ['teleport1', h(Teleport, { to: target }, 'teleport2')])
+        )
+      ]),
+      root
+    )
+    expect(serializeInner(target)).toMatchInlineSnapshot(
+      `"<div>teleport1<!--teleport start--><!--teleport end--></div>teleport2"`
+    )
+    expect(serializeInner(root)).toMatchInlineSnapshot(
+      `"<!--teleport start--><!--teleport end-->"`
+    )
+
+    render(null, root)
+    expect(serializeInner(target)).toBe('')
+  })
```

Comment:
This test case cannot reproduce #3623, because that issue only occurs in optimized mode. You can use the [template explorer](vue-next-template-explorer.netlify.app) to cooperate with you to find the minimal repro and use it as a test case

Reply:
Let me know if you need help, and of course, thank you for this PR

Reply:
I will ask you on WeChat later.